### PR TITLE
Added GCP Specific Pre-requisite setup

### PR DESCRIPTION
**Pull request by malscent**
Added GCP Post Processing Hook
Added a Wait option to wait to complete until after N nodes are in the cluster and healthy

### CODE REVIEW

**Review by tahmmee — COMMENTED** [review]
Looks good!  Some thoughts here on debugging in case something goes wrong.

File: compose-files/UbuntuClusterCompose.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ services:
     networks:
       - cb-net-ubuntu
     container_name: ubuntu-clusterhost
-    command: --version 6.6.1 -u couchbase -p foo123! -ch clusterhost -d -os UBUNTU -r
+    command: --version 6.6.1 -u couchbase -p foo123! -ch clusterhost -d -os UBUNTU -r -w 3
```
[the remark this answers is not in the document]
Comment (malscent) [reply compose-files/UbuntuClusterCompose.yml:15]: Good call

File: installers/installer.sh [hunk 2]
```diff
@@ -78,9 +78,29 @@ function __debian_prerequisites() {
     __ubuntu_prerequisites "$1"
 }
 
+function __get_gcp_metadata_value() {
+    curl --retry 5 -s -f -H "Metadata-Flavor: Google" "http://metadata/computeMetadata/v1/$1"
```
[the remark this answers is not in the document]
Comment (malscent) [reply installers/installer.sh:82]: Honestly,  it's a metadata service on a GCP instance,  I'm not sure how likely this is to fail.  There may be a race condition i guess where if this startup script is ran before the meta data service starts on the instance?  But I'm not sure how likely that is honestly.

File: compose-files/UbuntuClusterCompose.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (malscent) [reply compose-files/UbuntuClusterCompose.yml:15]: Fixed and changed to --wait-nodes to be more clear. Thanks @tahmmee

File: installers/installer.sh [hunk 3]
```diff
@@ -92,8 +112,32 @@ function __install_prerequisites() {
...
+    if [[ "$env" == "GCP" ]]; then
+        echo "Running GCP Prequisites"
+        ACCESS_TOKEN=$(__get_gcp_metadata_value "instance/service-accounts/default/token" | awk -F\" '{ print $4 }')
+        PROJECT_ID=$(__get_gcp_metadata_value "project/project-id")
+        EXTERNAL_IP=$(__get_gcp_metadata_value "instance/network-interfaces/0/access-configs/0/external-ip")
+        CONFIG=$(__get_gcp_attribute_value "runtime-config-name")
+        EXTERNAL_IP_VAR_PATH=$(__get_gcp_attribute_value "external-ip-variable-path")
+        SUCCESS_STATUS_PATH="$(__get_gcp_attribute_value "status-success-base-path")/$(hostname)"
+        FAILURE_STATUS_PATH="$(__get_gcp_attribute_value "status-failure-base-path")/$(hostname)"
+        NODE_PRIVATE_DNS=$(__get_gcp_metadata_value "instance/hostname")
```
[the remark this answers is not in the document]
Comment (malscent) [reply installers/installer.sh:141]: Added Log Statements for each one

**Pull request merged**